File: week5/simplepublic.cpp
#include<iostream>
using namespace std;
class Superclass{


    public:
        int a = 10;
    private:
        int b = 20;
    protected:
        int c = 30;
    public:
        Superclass(){
            cout << "In super class" << endl;
            cout << "public:" << a << endl;
            cout << "private:" << b <<  endl;
            cout << "protected:" << c << endl;
        }
        int getb()
        {
            return b;  //private variable 'b' accessible with in the class
        }
};
class Subclass : public Superclass{
    public:
        Subclass(): Superclass(){
            cout << "In sub class" << endl;
            cout << "public:" << a << endl;
            cout << "private:" << "private member cannot be accessed" <<  endl;
            cout << "protected:" << c << endl;
        }
        int getc()
        {
            return c;  //protected variable accessible within the class and  inherited class
        }
};
int main()
{
    Subclass obj;
    cout << "In main(): Subclass inherits public Superclass" << endl;
    cout << "public:" << obj.a << endl;
    // cout << "public:" << obj.b << endl;// 'int Superclass::b' is private within this context
    // cout << "public:" << obj.c << endl; // 'int Superclass::c' is protected within this context
    cout << "private:" << obj.getb() <<  endl;
    cout << "protected:" << obj.getc() << endl;


    return 0;
}

File: week5/Simplepublic.java
import pack1.Superclass;


class Subclass extends Superclass {
    public Subclass() {
        super();
        System.out.println("In sub class");
        System.out.println("public:" + a);
        System.out.println("private:" + "private member cannot be accessed");
    }


    int getc() {
        return c; // protected variable accessible within the class and inherited class
    }
}


public class Simplepublic {
    public static void main(String[] args) {
        Subclass obj = new Subclass();
        System.out.println("In main(): Subclass inherits public Superclass");
        System.out.println("public:" + obj.a);
        // System.out.println("private:" + obj.b); // b has private access in Superclass
        // System.out.println("protected:" + obj.c); // c has protected access in
        // Superclass


        System.out.println("private:" + obj.getb());
        System.out.println("protected:" + obj.getc());
    }
}
